cli/test/test-diff-utils.mjs

```
@@ -0,0 +1,107 @@
+await test('preserves trailing-newline split semantics', () => {
+  // 'a\n' splits to ['a', ''] — both pieces flow through unchanged.
+  const result = diffLines('a\n', 'a\n')
+  assertDeepEquals(result, [
+    { kind: 'eq', text: 'a' },
+    { kind: 'eq', text: '' },
+  ])
+})
+
+if (testsFailed > 0) {
+  console.error(`\n❌ ${testsFailed} test(s) failed`)
+  process.exit(1)
```

ESLint

[error] Unexpected use of the global variable 'process'. Use 'require("process")' instead.

(node/prefer-global/process)